ontology.ttl

```diff
@@ -256,7 +256,8 @@ owl:versionIRI rdf:type owl:AnnotationProperty .
                          [ rdf:type owl:Restriction ;
                            owl:onProperty :value ;
                            owl:allValuesFrom xsd:boolean
-                         ] .
+                         ]  ;
+       rdfs:equivalentClass xsd:boolean .
```

Comment:
It does not makes sense that it both `rdfs:equivalentClass xsd:boolean . and ` `..  owl:allValuesFrom xsd:boolean ..` (i.e. has value xsd:boolean).

This Float here is in fact Float Parameter. I suppose the difference between Float Parameter and float is that Float Parameter can have some additional restrictions.

So, we should just always return  Float Parameter and not xsd:float in nb2workflow.

How does it sound @burnout87 @dsavchenko ?

Reply:
> > It does not makes sense that it both rdfs:equivalentClass xsd:boolean . and ..  owl:allValuesFrom xsd:boolean .. (i.e. has value xsd:boolean).
> 
> They in principle represent two different types of relations: the first an equivalence between the two types (eg `Float` and `float`), whereas the second is one of the restriction to be applied. So I think it makes sense to have them both

Equivalence presumably implies that the same restriction implies to xsd:boolean, right?  
If so it is not really what  xsd:boolean is meant to have, as it is introduced by xsd.

> 
> > So, we should just always return Float Parameter and not xsd:float in nb2workflow.
> 
> At the moment the `xsd` type is assigned in the situation where for a declared parameter (eg `dT` in Fermi [oda-hub/renku-aqs#31 (comment)](https://github.com/oda-hub/renku-aqs/issues/31#issuecomment-1568156900)) no comment is provided. Otherwise it should have been explicitly declared as such if certain additional restrictions are desired.

Right, and as we use it now, it does not make sense to use xsd there since it's weird to define :value and various other restriction on types out of our ontology. It's better to always refer to our own types, like oda:Float, and express their relation with xsd:float through :value.

Reply:
ok then I will apply this approach in `nb2workflow`

Reply:
it's good to check-in with @dsavchenko since he added this part to the ontology

Reply:
This way we will end up with no changes in the ontology, which is also OK if hb2workflow will always return own types
The attempt to add this relation was made following https://github.com/oda-hub/renku-aqs/issues/31#issuecomment-1557382388. Changes to nb2workflow will eliminate this issue

Reply:
ok, thanks. Yes, in principle, because of the open world principle in this kind of kg, we could create a property with domain in xsd:float . But creating a property :value which has domain and range xsd:float  is just strange and self-referrential. It also would allow infinite chain of links like xsd:float :value xsd:float :value with unclear meaning. So let's not.

If the whole change is in nb2workflow, we can close this PR here.

Reply:
I agree,let's focus on changing the behavior of `nb2workflow`